receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,362 @@
+	if c.alerts != nil {
+		if err := c.alerts.Start(ctx, host); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	// If we have a logging receiver start log collection
+	if c.logs != nil {
+		if err := c.logs.Start(ctx, host); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+// Shutsdown the combined MongoDB Atlas Logs and Alert Receiver
+func (c *combindedLogsReceiver) Shutdown(ctx context.Context) error {
+	var errs error
+
+	// If we have an alerts receiver shutdown alerts collection
+	if c.alerts != nil {
+		if err := c.alerts.Shutdown(ctx); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	// If we have a logging receiver shutdown log collection
+	if c.logs != nil {
+		if err := c.logs.Shutdown(ctx); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func (s *receiver) createProjectMap() (map[string]*Project, error) {
+	if len(s.cfg.Logs.Projects) < 1 {
+		return nil, errors.New("no projects specified")
+	}
+
+	// create a map of Projects
+	projects := make(map[string]*Project)
+
+	for _, project := range s.cfg.Logs.Projects {
+		projects[project.Name] = project
+	}
+
+	return projects, nil
+}
+
+// ParseHostName parses out the hostname from the specified cluster host
+func ParseHostName(s string) []string {
+	var hostnames []string
+
+	// first check to make sure string is of adequate size
+	if len(s) < 1 {
+		return []string{}
+	}
+
+	// create an array with a comma delimiter from the original string
+	tmp := strings.Split(s, ",")
+	for _, t := range tmp {
+
+		u, err := url.Parse(t)
+		if err != nil {
+			fmt.Printf("Error parsing out %s", t)
+		}
+
+		// separate hostname from scheme and port
+		host, _, err := net.SplitHostPort(u.Host)
+		if err != nil {
+			// the scheme prefix was not added on to this string
+			// thus the hostname will have been placed under the "Scheme" field
+			hostnames = append(hostnames, u.Scheme)
+		} else {
+			// hostname parsed successfully
+			hostnames = append(hostnames, host)
+		}
+
+	}
+
+	return hostnames
+}
+
+// KickoffReceiver spins off functionality of the receiver from the Start function
+func (s *receiver) KickoffReceiver(ctx context.Context) {
+	resource := resourceInfo{start: strconv.Itoa(int(time.Now().Unix())), end: ""}
+	cfgProjects, err := s.createProjectMap()
+	if err != nil {
+		s.log.Error("Failed to create project map", zap.Error(err))
+	}
+
+	for {
+		// reterive all organizations listed in the MongoDB client
+		orgs, err := s.client.Organizations(ctx)
+		if err != nil {
+			s.log.Error("Error retrieving organizations", zap.Error(err))
+		}
+		// get all projects listed for each of the organizations
+		for _, org := range orgs {
+			projects, err := s.client.Projects(ctx, org.ID)
+			if err != nil {
+				s.log.Error("Error retrieving projects", zap.Error(err))
+			}
+
+			// filter out any projects not specified in the config
+			fp := filterProjects(projects, cfgProjects)
```

Comment:
Am I looking at a different package than you?

https://pkg.go.dev/go.mongodb.org/atlas/mongodbatlas#ProjectsService

Why not ask for the projects they've listed directly? Asking for all could be a scalability or security problem.